File: functional-test-suite/src/commonMain/resources/functions/connected/__connected__.py
class _connected:
    from ._support import Support

    def sum(i: int, j: int, bonus: int) -> int:
        # Separately, here's an example of where overriding name choices might
        # sometimes be handy.
        return _sum_of3(i, j, bonus)

    def prod(hidden: "_Hidden", j: int) -> int:
        return _connected.Support().prod(hidden.i, j)

    def length(string: str | None) -> int:
        return -1 if string is None else len(string)

File: functional-test-suite/src/commonMain/resources/functions/connected/_connected.cpp
#include "_connected.hpp"
#include "support.hpp"

namespace work {
namespace _connected {

std::int32_t sum(std::int32_t i, std::int32_t j, std::int32_t bonus) {
    return work::sumOf3(i, j, bonus);
}

std::int32_t prod(std::shared_ptr<Hidden> const& hidden, std::int32_t j) {
    Support support;
    return support.prod(hidden->get_i(), j);
}

int32_t length(temper::core::NullableParam<std::string> s) {
    return s.has_value ? s.value.size() : -1;
}

} // namespace _connected
} // namespace work
